src/app/Fake.Core.Trace/Trace.fs

```diff
@@ -232,11 +255,24 @@ let traceEndTaskUnsafe task = traceEndTaskUnsafeEx TagStatus.Success task
 /// Traces the end of a task
 [<System.Obsolete("Consider using traceTask instead and 'use' to properly call traceEndTask in case of exceptions. To remove this warning use 'traceEndTask'.")>]
 let traceEndTask task = traceEndTaskUnsafe task
-     
+
+/// Wrap functions in a 'use' of this function     
 let traceTask name description =
     traceStartTaskUnsafe name description
     asSafeDisposable (fun state -> traceEndTaskUnsafeEx state name)
 
+/// Traces a function execution
+/// If no exception is thrown then trace is marked as success
+/// Any exception thrown will result in a mark failed and exception re-thrown 
+let inline useWith (d:ISafeDisposable) f =
```

Comment:
yes it looks quite reasonable. I guess there are two versions, one where `MarkSuccess` is automatic and one where it is explicit.
I guess we could have either 2 functions or a boolean parameter. We could forward the `t` to `f` in order to allow `f` to change it's state internally.

(Reason is that calling nothing is equal to `inconclusive`)

Reply:
Because the signature of f would be different it had to be 2 functions.

I wonder if we really need the manually set one or if we can just have the automatic one